Replace deb packaging with AppImages.

pchote opened the pull request:
This PR applies https://github.com/OpenRA/OpenRAModSDK/pull/71 to our upstream mods and removes the now-superseded deb packaging.

Fixes #2585.
Fixes #12257.
Closes #9201.
Closes #13374 (run the AppImage with `--server`).
Closes #14894.

The generated AppImages have a couple of differences from the SDK versions:
1.  We now rebuild the icon cache as well as the mime cache on first run to fix the broken launcher icons (we'll want to backport this to the SDK).
2. We generate metadata for version updates, but the `master.openra.net/appimagecheck` end point is currently a placeholder that upgrades all mods to TD.

https://github.com/pchote/OpenRA/releases has test packages for the different release types.

Some notes:
* On a clean system players just need to download the appimages for the mods they want, make them executable, and then run them.
* The images should run on any reasonable Linux distro (>= CentOS 6) but, because they don't package mono, some users will first need to set up http://www.mono-project.com/download/stable/#download-lin.
* System integration for launcher shortcuts and protocol handlers requires the optional `appimaged` daemon (download from https://github.com/AppImage/AppImageKit/releases/tag/continuous, make it executable, then run with the --install argument).
* The current/old(?) way to enable updates is to download the optional `AppImageUpdate` utility from https://github.com/AppImage/AppImageUpdate/releases/tag/continuous, rename/move it to `/usr/local/bin`, and make it executable.  Log out/in so that `appimaged` can detect it, then right click on the launcher icon to see the update option.  The upstream AppImage system integration tools appear to be undergoing a rethink and rewrite so this will probably improve in the future.
* I have made sure that `make install` and `make uninstall` continue to work, so downstream packaging should be minimally affected.
* I have tested the system integration on Gnome (2 & 3) and Unity, but none of the other mainstream WMs.
* The download page and version metadata files will need to be reworked before we can push a release with these changes.

pchote committed:
Remove .deb package generation.

pchote committed:
Simplify mime type handling.

pchote committed:
Standardize on {MODID} in template files.

pchote committed:
Polish crash dialog code and UI.

pchote committed:
Fix appdata uninstallation.

pchote committed:
Rework icon installation.

abcdefg30 commented on packaging/linux/AppRun.in:
```diff
@@ -0,0 +1,36 @@
+
+if mono_missing_or_old; then
+  if command -v zenity > /dev/null
+  then
```
We moved the `then` on the same line as the `if`statement everywhere else.

abcdefg30 commented on packaging/linux/buildpackage.sh:
```diff
@@ -1,60 +1,136 @@
+	sed "s/{MODID}/${MOD_ID}/g" openra.appimage.in | sed "s/{TAG}/${TAG}/g" | sed "s/{MODNAME}/${DISPLAY_NAME}/g" > openra-mod.temp
+	install -m 0755 openra-mod.temp "${APPDIR}/usr/bin/openra-${MOD_ID}"
+
+	sed "s/{MODID}/${MOD_ID}/g" openra-server.appimage.in  > openra-mod-server.temp
```
Double whitespace in front of the `>`.

pchote committed:
Add AppImage packaging for Linux.

pchote commented:
Fixed and rebased.  Updated test images are available from https://github.com/pchote/OpenRA/releases/tag/pkgtest-20180502 if anyone wants to confirm that everything still works.

abcdefg30 commented:
`View FAQ` doesn't work anymore.

pchote committed:
Adjust zenity crash dialog.

This removes the "View FAQ" button, which does not work in the AppImage
enivronment.  It also replaces the inconsistent question mark icon with
a more appropriate stop symbol, and makes the wording consistent across
the zenity, terminal, and native macOS dialogs.

pchote committed:
Suppress GTK transient parent warning.

pchote commented:
I spent a bit of time on that, but couldn't get to the bottom of what exactly in the AppImage environment was breaking `xdg-open`.  The inconsistencies in the crash dialog (question mark icon, FAQ button but no wiki button, differences in text) always low-level bugged me, so I decided that a better solution was to change it to a proper `--error` dialog:

![crash](https://user-images.githubusercontent.com/167819/39655820-bdd73724-4ff3-11e8-9436-ba222c5db7d1.png)

Updated test images: https://github.com/pchote/OpenRA/releases/tag/pkgtest-20180504-2

pchote committed:
Extract error messages to a variable to reduce duplication.

pchote committed:
Add kdialog support for KDE.

pchote commented:
Two new commits reduce duplication and add graphical dialogs on KDE:
![screen shot 2018-05-05 at 00 02 40](https://user-images.githubusercontent.com/167819/39656536-897ac81a-4ff8-11e8-8f4b-7a4a2687043d.png)
![screen shot 2018-05-05 at 00 03 47](https://user-images.githubusercontent.com/167819/39656539-8c4d547c-4ff8-11e8-959d-953189ce7b79.png)


Updated (again) test images: https://github.com/pchote/OpenRA/releases/tag/pkgtest-20180504-3

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/470c5b862c7cc38d1e3a6e8625a6fada302f2efa...027927eda7520298a78730619edc417f6202b995)

TheAssassin commented:
> We now rebuild the icon cache as well as the mime cache on first run to fix the broken launcher icons (we'll want to backport this to the SDK).

See https://github.com/AppImage/AppImageKit/issues/653. Feel free to add your two cents to the discussion. The more people push the issue, the more likely it is that a solution will be implemented soon.
